### PR TITLE
Revise Split operation reference implementation

### PR DESCRIPTION
**Pull request by ggalieroc**
Split operation class refactored.

Details:
- Modified node validation checks in `validate_and_infer_types` method
- Added checks for host tensor inputs and outputs
- Added type_prop unit tests with invalid axis input and negative axis value
- Enabled CPU backend tests

Tickets: 45937, 45935, 37481

### CODE REVIEW

File: ngraph/core/src/op/split.cpp [hunk 1]
```diff
@@ -142,6 +138,9 @@ namespace split
 bool op::v1::Split::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const
 {
     NGRAPH_OP_SCOPE(v1_Split_evaluate);
+    NGRAPH_CHECK(this,
```
Comment (ilyachur) [inline ngraph/core/src/op/split.cpp:141]: It is not correct. In this case `NGRAPH_CHECK` doesn't check your condition, it checks: `!this`
Please use:
```
NODE_VALIDATION_CHECK(this, validate_host_tensor_vector(outputs, m_num_splits) &&
                     validate_host_tensor_vector(inputs, 2));
```
or
```
NGRAPH_CHECK(validate_host_tensor_vector(outputs, m_num_splits) &&
                     validate_host_tensor_vector(inputs, 2));
```

Comment (ggalieroc) [reply ngraph/core/src/op/split.cpp]: Good catch. Done

**Comment by blesniewski** [comment]
In your PR for Split's spec, you changed the default type of `num_splits` to int, but split.hpp suggests unsigned integer type (size_t to be precise). Can we rectify this somehow? I would suggest editing the spec, as size_t seems to be a good fit for this attribute.

**Review by blesniewski — APPROVED** [review]

File: ngraph/core/src/op/split.cpp [hunk 2]
```diff
@@ -36,28 +36,24 @@ bool ngraph::op::v1::Split::visit_attributes(AttributeVisitor& visitor)
...
     const auto axis_input = get_constant_from_source(input_value(1));
-    if (axis_input && data_ps.rank().is_static())
+    if (axis_input && data_rank.is_static())
     {
         auto axis = axis_input->cast_vector<int64_t>()[0];
-
-        const auto data_rank = get_input_partial_shape(0).rank();
         axis = ngraph::normalize_axis(this, axis, data_rank);
 
         if (data_ps[axis].is_static())
```
Comment (mitruska) [inline ngraph/core/src/op/split.cpp:59]: As the `num_splits` is an attribute, what do you think about setting at least upper bound for the output dynamic dimension? Should it work with dynamic interval dims?

https://github.com/openvinotoolkit/openvino/blob/358f3bcba418fe7f16d236e0aa7ae4094cc9649c/ngraph/core/src/op/split.cpp#L70-L75

Comment (lazarevevgeny) [reply ngraph/core/src/op/split.cpp:59]: May be it make sense to add `operator/` for the `Dimension` class (the interval logic looks appropriate for the division operation if we perform floor rounding). If we add it then we can write `each_output_shape[axis] = dimension_at_axis / m_num_splits;` regardless of the static/dynamic nature of the input.

@jane-intel , what do you think about this?

Comment (ggalieroc) [reply ngraph/core/src/op/split.cpp:59]: I have added few changes related to @mitruska comment. The behavior now is:

- if the upper bound of the interval is less than `num_splits` throw exception
- if dimension does not have upper bound, only lower bound is divided
- if dimension has lower and upper bounds, both are divided by `num_splits`

**Review by lazarevevgeny — APPROVED** [review]

File: ngraph/core/src/op/split.cpp [hunk 3]
```diff
@@ -66,16 +62,34 @@ void op::v1::Split::validate_and_infer_types()
...
+                " and ",
+                m_num_splits);
+
+            auto dim_interval_at_axis_min =
+                static_cast<int64_t>(dim_interval_at_axis.get_min_val() * (1.0f / m_num_splits));
+            auto dim_interval_at_axis_max = dim_interval_at_axis.get_max_val();
+            if (dim_interval_at_axis.has_upper_bound())
+            {
+                dim_interval_at_axis_max =
+                    static_cast<int64_t>(dim_interval_at_axis_max * (1.0f / m_num_splits));
```
Comment (jane-intel) [inline ngraph/core/src/op/split.cpp:90]: num_splits could be equal to zero -- not from the specification but from the code I read.

File: ngraph/core/src/op/split.cpp (hunk 2 above)
Comment (jane-intel) [reply ngraph/core/src/op/split.cpp:59]: I would prefer having operator/ defined for `Dimension`. We already have several operations that require it (like Interpolate)

File: ngraph/core/src/op/split.cpp (hunk 3 above)
Comment (ggalieroc) [reply ngraph/core/src/op/split.cpp:95]: I added a node validation check on `m_num_splits` attribute.

**Review by mitruska — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
Please rebase to the latest master in order to avoid any problems after the merge.

Sorry for inconvenience.

**Pull request merged**